File: back/app/forms/session_form.py
from app.constants.sessions import MAX_TITLE_LENGTH
from app.lib.mixin.validatable import Validatable, model_name
from app.validations.blank_validation import BlankValidation
from app.validations.max_length_validation import MaxLengthValidation
from app.lib.mixin.validatable import validate


@model_name('session')
@validate('title', BlankValidation)
@validate('title', MaxLengthValidation, limit=MAX_TITLE_LENGTH)
class SessionForm(Validatable):
    def __init__(self, **params):
        super().__init__()
        self.title = params.get('title')

File: front/src/models/forms/session-form.ts
import ApplicationForm from '@/models/forms/application-form';
import { blankValidation, maxLengthValidation } from '@/lib/validations';
import { MAX_TITLE_LENGTH } from '@/constants/sessions';
import { NewSessionRequest } from '@/models/interfaces/api/sessions';

export default class SessionForm extends ApplicationForm {
  token!: string;
  title!: string;

  constructor({ token = '', title = '' }: { token?: string; title?: string } = {}) {
    super();
    this.token = token;
    this.title = title;
  }

  initRules(): void {
    this.setTitleRules();
    this.setTokenRules();
  }

  createParams(): NewSessionRequest {
    return {
      title: this.title || ''
    };
  }

  private setTitleRules(): void {
    this.setRules(['Session', 'title'], (messageKey: string) => [
      blankValidation(messageKey),
      maxLengthValidation(messageKey, MAX_TITLE_LENGTH)
    ]);
  }

  private setTokenRules(): void {
    this.setRules(['Session', 'token'], (messageKey: string) => [blankValidation(messageKey)]);
  }
}
